core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts

```diff
@@ -186,15 +202,34 @@ export class DotPageStateService {
         return this.dotPageRenderService.get(options, extraParams).pipe(
             catchError((err) => this.handleSetPageStateFailed(err)),
             take(1),
-            map((page: DotPageRenderParameters) => {
+            switchMap((page: DotPageRenderParameters) => {
```

Comment:
Suggestion:

```typescript
    /**
     * Get the page from API, set local state and return DotPageRenderState
     *
     * @param {DotPageRenderOptions} options
     * @returns {Observable<DotPageRenderState>}
     * @memberof DotPageStateService
     */
    requestPage(options: DotPageRenderOptions): Observable<DotPageRenderState> {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const { url, ...extraParams } = this.dotRouterService.queryParams;

        return this.getFavFromPage(options.url).pipe(
            switchMap((fav: boolean) => {
                return this.dotPageRenderService.get(options, extraParams).pipe(
                    catchError((err) => this.handleSetPageStateFailed(err)),
                    take(1),
                    map((page: DotPageRenderParameters) => {
                        if (page) {
                            const pageState = new DotPageRenderState(
                                this.getCurrentUser(),
                                page,
                                fav
                            );
                            this.setCurrentState(pageState);

                            return pageState;
                        }

                        return this.currentState;
                    })
                );
            })
        );
    }

    private getFavFromPage(url: string): Observable<boolean> {
        return this.dotESContentService
            .get({
                itemsPerPage: 10,
                offset: '0',
                query: `+contentType:DotFavoritePage +DotFavoritePage.url_dotraw:${url}`
            })
            .pipe(
                take(1),
                map((response: ESContent) => response.resultsSize > 0)
            );
    }
```

Reply:
that won't work, I need to do the fetch of `dotPageRenderService` to get all attributes like language, device, host and Uri, and not all are present in `options: DotPageRenderOptions`